qiskit/providers/backend.py

```diff
@@ -229,3 +233,233 @@ class can handle either situation.
+    """Abstract class for Backends
+
+    This abstract class is to be used for all Backend objects created by a
+    provider. This version differs from earlier abstract Backend classes in
+    that the configuration attribute no longer exists instead attributes
+
+    The ``options`` attribute of the backend is used to contain the dynamic
+    user configurable options of the backend. It should be used more for
+    runtime options that configure how the backend is used. For example,
+    something like a ``shots`` field for a backend that runs experiments which
+    would contain an int for how many shots to execute.
+
+    If migrating a provider from :class:`~qiskit.providers.BackendV1` or
+    :class:`~qiskit.providers.BaseBackend` one thing to keep in mind is for
+    backwards compatibility you might need to add a configuration method that
+    will build a :class:`~qiskit.providers.models.BackendConfiguration` object
+    and :class:`~qiskit.providers.models.BackendProperties` from the attributes
+    defined in this class for backwards compatibility.
+    """
+
+    version = 2
+
+    def __init__(self, provider, **fields):
+        self._options = self._default_options()
+        self._provider = provider
+        self._coupling_map = None
+        if fields:
+            for field in fields:
+                if field not in self._options.data:
+                    raise AttributeError("Options field %s is not valid for this backend" % field)
+            self._options.update_config(**fields)
+
+    @property
+    def instructions(self) -> List[Gate]:
+        """A list of Gate instances that the backend supports."""
+        return list(self.target.gates)
+
+    @property
+    def instructions_names(self) -> List[str]:
+        """A list of Gate names that the backend supports."""
+        return list(self.target.gate_names)
+
+    @property
+    @abstractmethod
+    def target(self):
+        """A :class:`qiskit.transpiler.Target` object for the backend."""
+        pass
+
+    @property
+    @abstractmethod
+    def num_qubits(self) -> int:
+        """Return the number of qubits the backend has."""
+        pass
+
+    @property
+    def coupling_map(self):
+        """Return the :class:`~qiskit.transpiler.CouplingMap` object"""
+        return self.target.coupling_map()
+
+    @property
+    def instruction_durations(self):
+        """Return the :class:`~qiskit.transpiler.InstructionDurations` object."""
+        return self.target.durations()
+
+    @property
+    @abstractmethod
+    def conditional(self) -> bool:
+        """Return bool whether the target can execute gates with classical
+        conditions."""
+        pass
+
+    @property
+    @abstractmethod
+    def max_shots(self) -> int:
+        """Return the maximum number of shots supported by the backend.
+
+        If there is no limit this will return None
+        """
+        pass
+
+    @property
+    @abstractmethod
+    def max_experiments(self):
```

Comment:
This was supposed to be changed to `max_circuits` at some point.

Reply:
Sure, I'll rename it here

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/5885/commits/31eedbca25ccde9c8c9b916250f7f241081918e3